src/Propel/Runtime/Map/DatabaseMap.php

```diff
@@ -134,6 +139,38 @@ public function addTableFromMapClass(string $tableMapClass): TableMap
         return $this->getTable($table->getName());
     }
 
+    /**
+     * Dump table maps. Used during configuration generation.
+     *
+     * @return \Propel\Runtime\Map\TableMapDump A dump that can be loaded again with {@link DatabaseMap::loadMapsFromDump()}
```

Comment:
This looks like array<mixed, ..> instead of a object return though.

Reply:
Yes, good that you mention it, I am not sure how much I like that. Apparently psalm allows type aliases, which `\Propel\Runtime\Map\TableMapDump` is, I put the declaration in the docblock of the class:
```php
* @psalm-type \Propel\Runtime\Map\MapType = 'tablesByName' | 'tablesByPhpName'
 * @psalm-type \Propel\Runtime\Map\TableMapDump array<\Propel\Runtime\Map\MapType, array<string, class-string<\Propel\Runtime\Map\TableMap>>>
 ```
On one hand, it is nice, because we can make sure that the load and dump methods always use the exact same type, and through an import with `@psalm-import-type`, it can be used in the `StandardServiceContainer` too. So I think in general this is the "correct" approach.
On the other hand, a lot about it seems confusing. I had to give the alias a name that looks like a namespaced type (otherwise it wouldn't work with phpcs, which changed "TableMapDump" to "\Propel\Runtime\Map\TableMapDump", causing a no such type error). Also IDE support for type aliases is not (yet) what it could be, at least in VS Code and Eclipse. Overall, I have the impression that type aliases as a concept is still rather new in PHP, which may or may not keep us from using it. What do you think?

Reply:
I think the main docblock (return) should reflect trivial reality (array)
The on top specific phpstan and psalm tags can contain the "new" concepts.

Reply:
Like this then?
```php
    /**
     * Dump table maps. Used during configuration generation.
     *
     * @psalm-return \Propel\Runtime\Map\TableMapDump
     *
     * @return array A dump that can be loaded again with {@link DatabaseMap::loadMapsFromDump()}
     */
```